src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -73,6 +73,20 @@ public bool IsProcessingValue
                     return false;
                 }
 
+                if (PropertyInitialized)
```

Comment:
This simple method has grown up to be very ugly. This method and dependencies need to be refactored (not necessarily in this PR).